sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/CastSuite.scala

```diff
@@ -664,6 +665,25 @@ class CastSuite extends CastSuiteBase {
     }
   }
 
+  test("SPARK-35820: Support cast DayTimeIntervalType in different fields") {
+    val ym = Duration.ofSeconds(12345678L, 123456789)
```

Comment:
This is not year-month. Please, rename `ym`.

Reply:
> This is not year-month. Please, rename `ym`.

Done, sorry for miss this